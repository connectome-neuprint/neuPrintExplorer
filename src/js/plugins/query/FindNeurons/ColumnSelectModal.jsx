import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import makeStyles from '@mui/styles/makeStyles';
import Modal from '@mui/material/Modal';
import Button from '@mui/material/Button';

const useStyles = makeStyles((theme) => ({
  paper: {
    position: 'absolute',
    width: '95%',
    height: '95%',
    backgroundColor: theme.palette.background.paper,
    border: '2px solid #000',
    boxShadow: theme.shadows[5],
    padding: theme.spacing(2, 4, 3),
    top: `50%`,
    left: `50%`,
    transform: `translate(-50%, -50%)`,
  },
}));

export default function ColumnSelectModal({ callback, dataset }) {
  const [open, setOpen] = useState(false);
  const [disabled, setDisabled] = useState(true);
  const classes = useStyles();

  const handleOpen = () => {
    setOpen(true);
    if (callback) {
      callback();
    }
  };

  const selectorSrc = `/public/roi_selectors/${dataset}/column_selection_ui.html`;

  useEffect(() => {
    // if the file at selectorSrc can be fetched, then activate the button
    fetch(selectorSrc, {method: 'get'}).then(response => {
      if (response.ok) {
        setDisabled(false);
      }
    });
  },[selectorSrc]);


  return (
    <div>
      <Button color="primary" variant="outlined" onClick={handleOpen} disabled={disabled}>
        Column Selection
      </Button>
      <Modal
        open={open}
        onClose={() => setOpen(false)}
        aria-labelledby="simple-modal-title"
        aria-describedby="simple-modal-description"
      >
        <div className={classes.paper}>
          <Button
            style={{ position: 'absolute', top: '5px', right: '5px' }}
            variant="outlined"
            onClick={() => setOpen(false)}
          >
            ✖︎ Close
          </Button>
          <h2>ME(R) column selector</h2>
          <p>
            Hold down your mouse button, drag it over the columns you are interested in, to
            highlight them. Once highlighted, copy the ids from the text area, close the modal, and
            paste them into the &quot;Input/Output Brain Regions&quot; selection menus.
          </p>

          <iframe
            style={{ border: 'none' }}
            width="100%"
            height="100%"
            src={selectorSrc}
            title="Column selection"
          />
        </div>
      </Modal>
    </div>
  );
}

ColumnSelectModal.propTypes = {
  callback: PropTypes.func,
  dataset: PropTypes.string.isRequired
};

ColumnSelectModal.defaultProps = {
  callback: null,
};
