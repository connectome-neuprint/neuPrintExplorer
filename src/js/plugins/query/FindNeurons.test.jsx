import React from 'react';
import { render } from '../../tests/test-utils';

// import { ColorLegend } from 'plugins/MiniRoiHeatMap';
import { FindNeurons } from './FindNeurons';

const styles = { select: {}, clickable: {} };
const { actions, submit } = global;

const neoServerSettings = {
  get: () => 'http://example.com'
};

const component = (
  <FindNeurons
    availableROIs={['roiA', 'roiB', 'roiC']}
    dataSet="test"
    datasetstr="test"
    actions={actions}
    submit={submit}
    classes={styles}
    history={{ push: jest.fn() }}
    isQuerying={false}
    neoServerSettings={neoServerSettings}
    neoServer="testServer"
    superROIs={[]}
  />
);

describe('find neurons Plugin', () => {
  beforeEach(() => {
    submit.mockClear();
  });
  it('has name and description', () => {
    expect(FindNeurons.details.name).toBeTruthy();
    expect(FindNeurons.details.description).toBeTruthy();
  });
  it('renders correctly', () => {
    const { asFragment } = render(component);
    expect(asFragment()).toMatchSnapshot();
  });
  /*
  describe('when user clicks submit', () => {
    it('should return a query object and submit', () => {
      expect(button.props().onClick()).toEqual(undefined);
      expect(submit).toHaveBeenCalledTimes(1);

      // if neuron name/id is present add to parameters
      textField.props().onChange({ target: { value: 'abc' } });
      expect(button.props().onClick()).toEqual(undefined);
      textField.props().onChange({ target: { value: '123' } });
      expect(button.props().onClick()).toEqual(undefined);

      // if input/output rois present add to parameters
      textField.props().onChange({ target: { value: '' } });
      inputSelect.props().onChange([{ value: 'roiA' }]);
      outputSelect.props().onChange([{ value: 'roiB' }]);
      expect(button.props().onClick()).toEqual(undefined);

      // if neuron/segment filters present add to parameters
      limitNeuronsToggle.props().onChange();
      preThresholdField.props().onChange({ target: { value: 12 } });
      postThresholdField.props().onChange({ target: { value: 13 } });
      expect(button.props().onClick()).toEqual(undefined);

      expect(submit).toHaveBeenCalledTimes(5);
    });

    const query = {
      dataSet: 'test',
      queryString: '/npexplorer/findneurons',
      visType: 'SimpleTable',
      plugin: 'FindNeurons',
      visProps: { rowsPerPage: 25 },
      pm: {
        dataset: 'test',
        input_ROIs: [],
        output_ROIs: [],
        all_segments: false,
        statuses: []
      },
      title: 'Neurons with inputs in [] and outputs in []'
    };

    const apiResponse = {
      data: [
        [
          null,
          1,
          'KC-s',
          'Traced',
          '{"roiA":{"pre":22,"post":28},"roiB":{"pre":23,"post":31},"roiC":{"pre":45,"post":61}}',
          37325787,
          90,
          120,
          ['roiA', 'roiB', 'roiC'],
          true
        ]
      ],
      columns: [],
      debug: 'test'
    };

    it('should process returned results into data object', () => {

      const processedResults = FindNeurons.processResults(query, apiResponse, actions, submit);
      expect(processedResults).toEqual(
        expect.objectContaining({
          columns: [
            'id',
            'instance',
            'type',
            'status',
            'inputs (#post)',
            'outputs (#pre)',
            '#voxels',
            'roi breakdown',
            <div>
              roi heatmap <ColorLegend />
            </div>
          ],
          data: expect.arrayContaining([]),
          debug: 'test'
        })
      );
    });

    it('should return nothing if no data returned', () => {

      // if no data returned
      const processedResultsEmpty = FindNeurons.processResults(
        query,
        {
          columns: [],
          data: [],
          debug: 'test'
        },
        actions,
        submit
      );
      expect(processedResultsEmpty).toEqual({
        columns: [
          'id',
          'instance',
          'type',
          'status',
          'inputs (#post)',
          'outputs (#pre)',
          '#voxels',
          'roi breakdown',
          <div>
            roi heatmap <ColorLegend />
          </div>,
        ],
        data: [],
        debug: 'test',
        title: 'Neurons with inputs in [] and outputs in []'
      });

      // if rois selected should add roi count columns
      const queryWithRoisSelected = {
        dataSet: 'test',
        queryString: '/npexplorer/findneurons',
        visType: 'SimpleTable',
        plugin: 'FindNeurons',
        visProps: { rowsPerPage: 25 },
        pm: {
          dataset: 'test',
          input_ROIs: ['roiA'],
          output_ROIs: ['roiB'],
          all_segments: false,
          statuses: []
        },
        title: 'Neurons with inputs in [] and outputs in []'
      };
      const processedResultsWithRois = FindNeurons.processResults(
        queryWithRoisSelected,
        apiResponse,
        actions
      );
      expect(processedResultsWithRois).toEqual(
        expect.objectContaining({
          columns: [
            'id',
            'instance',
            'type',
            'status',
            'inputs (#post)',
            'outputs (#pre)',
            'roiA #post',
            'roiA #pre',
            'roiB #post',
            'roiB #pre',
            '#voxels',
            'roi breakdown',
            <div>
              roi heatmap <ColorLegend />
            </div>
          ],
          data: expect.arrayContaining([]),
          debug: 'test'
        })
      );
    });
  });

  describe('when user hits enter key', () => {
    it('should submit request', () => {
      const preventDefault = jest.fn();
      textField.props().onKeyDown({ keyCode: 13, preventDefault });
      expect(preventDefault).toHaveBeenCalledTimes(1);
      expect(submit).toHaveBeenCalledTimes(1);
    });
  });
  describe('when user inputs text or selects rois', () => {
    it('should change state', () => {
      actions.setQueryString.mockClear();

      // neuron name input
      textField.props().onChange({ target: { value: 'abc' } });
      expect(wrapper.find('FindNeurons').state('neuronInstance')).toBe('abc');

      // input rois
      inputSelect.props().onChange([{ value: 'roiA' }, { value: 'roiB' }]);
      expect(wrapper.find('FindNeurons').state('inputROIs')).toContainEqual('roiA');
      expect(wrapper.find('FindNeurons').state('inputROIs')).toContainEqual('roiB');
      expect(wrapper.find('FindNeurons').state('inputROIs').length).toBe(2);

      // output rois
      outputSelect.props().onChange([{ value: 'roiB' }, { value: 'roiC' }]);
      expect(wrapper.find('FindNeurons').state('outputROIs')).toContainEqual('roiC');
      expect(wrapper.find('FindNeurons').state('outputROIs')).toContainEqual('roiB');
      expect(wrapper.find('FindNeurons').state('outputROIs').length).toBe(2);
    });
  });
  */
});
