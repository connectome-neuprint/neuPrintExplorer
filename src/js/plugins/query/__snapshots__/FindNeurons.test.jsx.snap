// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`find neurons Plugin renders correctly 1`] = `
<DocumentFragment>
  <div>
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1tkmop-MuiFormLabel-root-MuiInputLabel-root"
      for="select-multiple-chip"
    >
      Neuron Instance, Type or BodyId (optional)
    </label>
    <div
      class="NeuronInputField-select-1 css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-2-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
        role="log"
      />
      <div
        class="asyncSelect__control css-13cymwt-control"
      >
        <div
          class="asyncSelect__value-container css-hlgwow"
        >
          <div
            class="asyncSelect__placeholder css-1jqq78o-placeholder"
            id="react-select-2-placeholder"
          >
            Type or Paste text for options
          </div>
          <div
            class="asyncSelect__input-container css-19bb58m"
            data-value=""
          >
            <input
              aria-activedescendant=""
              aria-autocomplete="list"
              aria-describedby="react-select-2-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class="asyncSelect__input"
              id="search-neuron-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="asyncSelect__indicators css-1wy0on6"
        >
          <span
            class="asyncSelect__indicator-separator css-1u9des2-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            class="asyncSelect__indicator asyncSelect__dropdown-indicator css-1xc3v61-indicatorContainer"
          >
            <svg
              aria-hidden="true"
              class="css-tj5bde-Svg"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <fieldset>
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1tkmop-MuiFormLabel-root-MuiInputLabel-root"
        for="select-multiple-chip"
      >
        Input Brain Regions
      </label>
      <div
        class="BrainRegionInput-select-2 css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
          role="log"
        />
        <div
          class="css-13cymwt-control"
        >
          <div
            class="css-hlgwow"
          >
            <div
              class="css-1jqq78o-placeholder"
              id="react-select-3-placeholder"
            >
              Select...
            </div>
            <div
              class="css-19bb58m"
              data-value=""
            >
              <input
                aria-activedescendant=""
                aria-autocomplete="list"
                aria-describedby="react-select-3-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-3-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="css-1wy0on6"
          >
            <span
              class="css-1u9des2-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-1946b2z-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <label
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-1946b2z-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              Match All
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
              >
                <span
                  class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
                    name="checked"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
                />
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              Any
            </div>
          </label>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          style="line-height: 0em;"
        >
          <svg
            aria-haspopup="true"
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-5 css-1umw9bq-MuiSvgIcon-root"
            data-testid="InfoIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-6h2zm0-8h-2V7h2z"
            />
          </svg>
        </div>
      </div>
    </fieldset>
    <fieldset>
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1tkmop-MuiFormLabel-root-MuiInputLabel-root"
        for="select-multiple-chip"
      >
        Output Brain Regions
      </label>
      <div
        class="BrainRegionInput-select-2 css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-4-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
          role="log"
        />
        <div
          class="css-13cymwt-control"
        >
          <div
            class="css-hlgwow"
          >
            <div
              class="css-1jqq78o-placeholder"
              id="react-select-4-placeholder"
            >
              Select...
            </div>
            <div
              class="css-19bb58m"
              data-value=""
            >
              <input
                aria-activedescendant=""
                aria-autocomplete="list"
                aria-describedby="react-select-4-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-4-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="css-1wy0on6"
          >
            <span
              class="css-1u9des2-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-1946b2z-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <label
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-1946b2z-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              Match All
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
              >
                <span
                  class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
                    name="checked"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
                />
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              Any
            </div>
          </label>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          style="line-height: 0em;"
        >
          <svg
            aria-haspopup="true"
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-5 css-1umw9bq-MuiSvgIcon-root"
            data-testid="InfoIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-6h2zm0-8h-2V7h2z"
            />
          </svg>
        </div>
      </div>
    </fieldset>
    <div
      class="MuiFormControl-root css-1iw3t7y-MuiFormControl-root"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
            />
          </span>
          <span
            class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
          />
        </span>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-1i3tsf3-MuiTypography-root"
          style="display: inline-flex;"
        >
          Allow all brain regions
        </h6>
      </label>
    </div>
    <div
      class="MuiFormControl-root css-1iw3t7y-MuiFormControl-root"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
            />
          </span>
          <span
            class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
          />
        </span>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-1i3tsf3-MuiTypography-root"
          style="display: inline-flex;"
        >
          Advanced input
        </h6>
      </label>
    </div>
    <div
      class="makeStyles-expandablePanel-7"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1808mag-MuiPaper-root-MuiAccordion-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <h3
          class="MuiAccordion-heading css-cy7rkm-MuiAccordion-heading"
        >
          <button
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1f1pi2l-MuiButtonBase-root-MuiAccordionSummary-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-yfrx4k-MuiAccordionSummary-content"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 css-1i3tsf3-MuiTypography-root"
              >
                Optional neuron/segment filters
              </h6>
            </span>
            <span
              class="MuiAccordionSummary-expandIconWrapper css-1wqf3nl-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
          </button>
        </h3>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-cwrbtg-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-1x6hinx-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-1i4ywhz-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiAccordion-region"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root makeStyles-nopad-8 css-1lj39kh-MuiAccordionDetails-root"
                >
                  <p>
                    Loading...
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Submit
    </button>
  </div>
</DocumentFragment>
`;
